fmriprep/workflows/bold/base.py

```diff
@@ -488,43 +494,14 @@ def init_func_preproc_wf(bold_file, has_fieldmap=False):
         workflow.connect([
             (initial_boldref_wf, bold_stc_wf, [
                 ("outputnode.skip_vols", "inputnode.skip_vols")]),
+            (bold_source, bold_stc_wf, [("out", "inputnode.bold_file")]),
             (bold_stc_wf, boldbuffer, [("outputnode.stc_file", "bold_file")]),
         ])
         # fmt:on
-        if not multiecho:
-            # fmt:off
-            workflow.connect([
-                (initial_boldref_wf, bold_stc_wf, [
-                    ("outputnode.bold_file", "inputnode.bold_file"),
-                ])
-            ])
-            # fmt:on
-        else:  # for meepi, iterate through stc_wf for all workflows
-            meepi_echos = boldbuffer.clone(name="meepi_echos")
-            meepi_echos.iterables = [
-                ("bold_file", bold_file),
-                ("name_source", bold_file),
-            ]
-            # fmt:off
-            workflow.connect([
-                (meepi_echos, bold_stc_wf, [("bold_file", "inputnode.bold_file")]),
-                (meepi_echos, boldbuffer, [("name_source", "name_source")]),
-            ])
-            # fmt:on
 
     # bypass STC from original BOLD in both SE and ME cases
-    elif not multiecho:  # SE and skip-STC
-        # fmt:off
-        workflow.connect([
-            (initial_boldref_wf, boldbuffer, [("outputnode.bold_file", "bold_file")]),
-        ])
-        # fmt:on
-    else:  # ME and skip-STC
-        # for meepi, iterate over all meepi echos to boldbuffer
-        boldbuffer.iterables = [
-            ("bold_file", bold_file),
```

Comment:
It looks like we had a ME/SE divergence here: For single-echo, we fixed up the header fields (matching qform/sform) but not for multi-echo. Will propose a fixup...

Reply:
The `validated_bold` `Select()` interface should now ensure that validated files are passed down for either ME or SE.